drivers/soundwire/stream.c

```diff
@@ -637,8 +637,32 @@ static int sdw_notify_config(struct sdw_master_runtime *m_rt)
 static int sdw_program_params(struct sdw_bus *bus, bool prepare)
 {
 	struct sdw_master_runtime *m_rt;
+	struct sdw_slave *slave;
 	int ret = 0;
 
+	list_for_each_entry(slave, &bus->slaves, node) {
+		u32 scale_index;
+		u32 addr1;
+		u8 base;
+
+		scale_index = sdw_slave_get_scale_index(slave, &base);
+		if (scale_index < 0)
+			return scale_index;
+
+		if (bus->params.next_bank) {
+			addr1 = SDW_SCP_BUSCLOCK_SCALE_B1;
+		} else {
+			addr1 = SDW_SCP_BUSCLOCK_SCALE_B0;
+		}
+		/* Program SDW_SCP_BUSCLOCK_SCALE */
+		ret = sdw_write_no_pm(slave, addr1, scale_index);
```

Comment:
careful here, not all peripherals support this register, and even worse not all peripherals support a dynamic clock scaling while audio is streaming.

so I think you need to make sure all devices are SDCA, we track this property somewhere.

And depending on the clock stale the frame size also needs to change accordingly. I don't see where the frame size is selected.

Reply:
For those don't support dynamic clock scaling, I guess we should always use the max clock, right?

Reply:
I think you should use the lower clock always, unless all components are SCDA.

Reply:
The main issue is that the default configuration for LNL+ supports 2 frequencies, but the RVPs and AIOC are all with SDCA devices.
The problem starts if someone starts using non-SDCA devices with LNL+, then we might change the clock. So we've got to guard against this.

For older platforms, there is only one frequency anyways so there's no choice.

Reply:
One easy solution is to always ignore default_col from ACPI and generate it at run time in the kernel. The default rows is correct.

Reply:
The higher clock is problematic on some designs, you may run into electrical/timing issues. So my recommendation is to ONLY use the higher clock starting with LNL and all SDCA setups.

Reply:
> careful here, not all peripherals support this register, and even worse not all peripherals support a dynamic clock scaling while audio is streaming.
> 
> so I think you need to make sure all devices are SDCA, we track this property somewhere.
> 

I check if sdca version > 0. Not sure what SDCA version supports. It seems to be SDCA 1.0, but rt722 which complies SDCA 0.8rc1 supports this.
```
			if (s_rt->slave->sdca_data.interface_revision == 0) {
				dev_err(&s_rt->slave->dev, "The Peripheral doesn't comply with SDCA\n");
				return -EINVAL;
			}
```

Reply:
I think all you need is 
````c
	if (!slave->id.class_id)
````

presumably all devices which set this bit in the ADR support the scaling registers, but better check with Cirrus Logic and Realtek